pkg/api/validation/validation.go

```diff
@@ -737,13 +755,14 @@ func ValidateMinionUpdate(oldMinion *api.Node, minion *api.Node) errs.Validation
 // Validate compute resource typename.
 // Refer to docs/resources.md for more details.
 func validateResourceName(str string) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
 	if !util.IsQualifiedName(str) {
-		return errs.ValidationErrorList{fmt.Errorf("invalid compute resource typename format %q", str)}
+		return append(allErrs, errs.NewFieldInvalid("resource typename", str, QualifiedNameErrorMsg))
```

Comment:
the first argument to NewFieldInvalid should be passed in to this function by the caller, or else it is not very useful.  Then you need to clarify  the "name" in the string.  Something like:

```
func validateResourceName(value string, field string) errs.ValidationErrorList {
    ...
    return errs.ValidationErrorList{NewFieldInvalid(field, value, "resource type " + QualifiedNameErrorMsg)}